docs/operators/deduper.md

```diff
@@ -0,0 +1,523 @@
+    which expired tuples are sent out by the Deduper.
+
+Attributes
+-----------------------
+
+-   ***Input port Attribute - input.TUPLE\_CLASS*** - Class or FQCN
+  -   Mandatory attribute
+  -   Tells the operator about the type of the incoming
+    tuple.
+
+
+
+Properties
+-----------------------
+
+-   ***keyExpression*** - String
+    -   Mandatory parameter.
+    -   The java expression to extract the key fields in the incoming tuple (POJO)
+
+
+-   ***timeExpression*** - String - (Time Based Deduper only)
+    -   The java expression to extract the time field in the incoming tuple (POJO).
+    -   In case there is no time field in the tuple, system time, when the tuple is processed, will be used.
+
+
+-   ***expireBefore*** - Long (Seconds) - (Time Based Deduper only)
+    -   This is the total time period during which a tuple stays in the system and blocks any other tuple with the same key.
+
+
+-   ***bucketSpan*** - Long (Seconds) - (Time Based Deduper only)
+    -   Mandatory parameter
+    -   This is the unit which describes how large a bucket can be. Typically this should be defined depending on the use case. For example, if we have expireBefore set to 1 hour, then typically we would be clubbing data in the order of minutes, so a bucketSpan of
+    around 1 minute or 5 minutes would make sense. Note that in this case, the entire data worth 1 minute or 5 minutes will expire as a whole. Setting it to 1 minute would make the number of time buckets in the system to be 1 hour / 1 minute = 60 buckets.  Similarly setting bucketSpan to 5 minutes would make number of buckets to be 12.
+    - Note that having too many or too less buckets could have a performance impact. If unsure, set the bucketSpan to be `~ sqrt(expireBefore)`. This way the number of buckets and bucket span are balanced.
+
+
+-   ***referenceInstant*** -  Long (Seconds) - (Time Based Deduper only)
+    -   The reference point from which to start the time which is use for expiry. Setting the referenceInstant to say, r seconds from the epoch, would initialize the start of expiry to be from that `instant = r`. The start and end of the expiry window periodically move by the span of a single bucket.
+
+
+-   ***numBuckets*** -  Integer - (Bounded Deduper only)
+    -   Optional parameter, but recommended to be provided by the user.
+    -   This is the number of buckets that need to be used for storing the keys of the incoming tuples.
+    -   Users can decide upon the proper value for this parameter by guessing the number of distinct keys in the application. An appropriate value would be sqrt(num distinct keys). In case, the number of distinct keys is a huge number, leave it blank so that the default value of 46340 will be used. The rationale for using this number is that sqrt(max integer) = 46340. This implies that the number of buckets used will roughly be equal to the size of each bucket, thus spreading the load equally among each bucket.
```

Comment:
Rewrite more compactly: A reasonable value is the square root of N where N is the estimated number of distinct keys. If omitted, the Java MAX_VALUE integer is used for N.